### PR TITLE
Move some `af` subcommands to newshell

### PR DESCRIPTION
**Pull request by ret2libc**
**DO NOT SQUASH**

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Move some analysis commands to newshell. See commits for a full list.

**Test plan**

Play with converted `af` commands + green CI.

**Closing issues**

...

### CODE REVIEW

**Comment by XVilka** [comment]
I think we should not squash those to be a good example for the future conversion.

File: librz/core/cmd_analysis.c [hunk 1]
```diff
@@ -9251,3 +9234,734 @@ RZ_IPI int rz_cmd_analysis(void *data, const char *input) {
...
+	if (mode == RZ_OUTPUT_MODE_JSON) {
+		pj_end(pj);
+		char *j = pj_drain(pj);
+		rz_cons_printf("%s\n", j);
+		free(j);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+static int outputmode2type(RzOutputMode mode) {
```
Comment (XVilka) [inline librz/core/cmd_analysis.c]: This maybe to put somewhere in `librz/core/core.c` or similar, you will need it in other command handlers too.

File: librz/core/cmd_analysis.c [hunk 2]
```diff
@@ -9251,3 +9234,734 @@ RZ_IPI int rz_cmd_analysis(void *data, const char *input) {
...
+	int type = outputmode2type(mode);
+	rz_analysis_var_list_show(core->analysis, fcn, kind, type, pj);
+	if (mode == RZ_OUTPUT_MODE_JSON) {
+		rz_cons_println(pj_string(pj));
+		pj_free(pj);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_analysis_function_vars_bp_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
```
Comment (XVilka) [inline librz/core/cmd_analysis.c]: What about putting a visual separator like `/// ---------------------------------------` or similar before these `RZ_IPI RzCmdStatus ...` function start?

File: librz/core/cmd_descs.h [hunk 3]
```diff
@@ -51,6 +51,47 @@ RZ_IPI RzCmdStatus rz_cmd_help_search_handler(RzCore *core, int argc, const char
 RZ_IPI int rz_cmd_help(void *data, const char *input);
 RZ_IPI RzCmdStatus rz_push_escaped_handler(RzCore *core, int argc, const char **argv);
 RZ_IPI int rz_cmd_ox(void *data, const char *input);
+RZ_IPI RzCmdStatus rz_analysis_function_blocks_list_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode);
```
Comment (XVilka) [inline librz/core/cmd_descs.h:54]: Same here - maybe it's better to separate the `RZ_IPI RzCmdStatus` functions from the rest?

Comment (ret2libc) [reply librz/core/cmd_descs.h:54]: this is auto-generated, no need for a separator or anything.

**Review by XVilka — APPROVED** [review]

**Review by thestr4ng3r — APPROVED** [review]

**Comment by XVilka** [comment]
@ret2libc please convert also those:
```
librz/core/cmd_analysis.c

8152:                   rz_core_cmd0(core, "aef@@F");
8175:                   rz_core_cmd0(core, "afr@@c:isq");
8216:           rz_core_cmd0(core, "af @@= `isq~[0]`");
8217:           rz_core_cmd0(core, "af @@f:entry*");
8220:           rz_core_cmd0(core, "af @@f:sym.*");
8221:           rz_core_cmd0(core, "af @@f:entry*");

```

**Comment by thestr4ng3r** [comment]
`afvb` and others now need the type argument to be quoted:
allowed before: ```afvb 16 argv const char **```
now:  ```afvb 16 argv "const char **"```
Do you think it should have `RZ_CMD_ARG_FLAG_LAST` set?

**Comment by XVilka** [comment]
@thestr4ng3r looks like already solved by @ret2libc

**Pull request merged**